.github/workflows/on-merge-to-main.yml

```diff
@@ -38,3 +38,103 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           TAG_PREFIX: v
           DEFAULT_BUMP: ${{ github.event.inputs.bump_type || 'patch' }}
+
+  build:
+    needs: versioning
+    if: needs.versioning.outputs.new_tag
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-latest]
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v4
+        with:
+          lfs: true
+          ref: ${{ needs.versioning.outputs.new_tag }}
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v4
+      - name: Setup Node.js
+        uses: actions/setup-node@v5
+        with:
+          node-version: 23
+          cache: 'pnpm'
+      - name: Install dependencies
+        run: pnpm install --shamefully-hoist
+        env:
+          OKTANA_NPM_TOKEN: ${{ secrets.OKTANA_NPM_TOKEN }}
+      - name: Build Electron app
+        run: pnpm run build
+        env:
+          OKTANA_NPM_TOKEN: ${{ secrets.OKTANA_NPM_TOKEN }}
+      - name: Package Electron app
+        run: pnpm run package
+        env:
+          OKTANA_NPM_TOKEN: ${{ secrets.OKTANA_NPM_TOKEN }}
+
+      # OS-specific uploads
+      - name: Upload macOS artifacts
+        if: matrix.os == 'macos-latest'
+        uses: actions/upload-artifact@v4
+        with:
+          name: macos-build-${{ needs.versioning.outputs.new_tag }}
+          path: |
+            bin/*.dmg
+            bin/*.zip
+
+      - name: Upload Windows artifacts
+        if: matrix.os == 'windows-latest'
+        uses: actions/upload-artifact@v4
+        with:
+          name: windows-build-${{ needs.versioning.outputs.new_tag }}
+          path: |
+            bin/*.exe
+            bin/*.zip
+
+      - name: Upload Linux artifacts
```

Comment:
I tried this AppImage and noticed it is not executable. We could add one more step here (or later, in `release`) like:
```yml
- name:  Make AppImage executable
  if: matrix.os == 'ubuntu-latest'
  run: chmod +x ./bin/*.AppImage
```

Reply:
Thanks, done! https://github.com/oktana-coop/v2/commit/8e6bd8d886c8503fc72dc058fa10ae9a3d85d7cb